extensions/amp-analytics/0.1/test/test-vendors-new.js

```diff
@@ -104,6 +109,34 @@ describes.realWin(
               config = analytics.config_;
               done();
             });
+
+            // Have to get service after analytics element is created
+            const variableService = variableServiceForDoc(doc);
+
+            window.sandbox
+              .stub(variableService, 'getMacros')
+              .callsFake(function() {
+                // Add all the macros in amp-analytics
+                const merged = {...this.macros_, ...elementMacros};
+
+                // Change the resolving function
+                const keys = Object.keys(merged);
+                for (let i = 0; i < keys.length; i++) {
+                  const key = keys[i];
+                  // Three opt params here because that is currently
+                  // the max number of params for any macro
+                  merged[key] = (opt_param, opt_param2, opt_param3) => {
+                    let params = [opt_param, opt_param2, opt_param3]
+                      .filter(val => val !== undefined)
+                      .join(',');
+                    if (params) {
+                      params = '(' + params + ')';
+                    }
+                    return `_${key.replace('$', '')}${params}_`;
```

Comment:
To confirm: 
analytics macros like `EXAMPLE(FAKE, param1, param2)` is replaced to `_EXAMPLE(_fake_, param1,param2)_`
global macros like `CLIENT_ID(param)` is replaced with `_client_id_`

To get the same behavior across all variables, can we please
1. Resolve convert analytics macro to lower case as well. 
2. Include the param value for global variables like CLIENT_ID. 

Feel free to address 2 in a separate PR. Thanks.

Reply:
Added 1 (which solved the double expansion problem actually!)
Will address 2 in a separate PR as part of this project.